### PR TITLE
Dev/hmm

### PR DESCRIPTION
**Pull request by mattbrown11**

### CODE REVIEW

File: angel_system/activity_hmm/simulate_data.py [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+             'Set timer to 30 seconds',
+             'Pour a small amount of water over the grounds in order to wet the grounds',
+             'Slowly pour the water over the grounds in a circular motion. Do not overfill beyond the top of the paper filter',
+             'Allow the rest of the water in the dripper to drain',
+             'Remove the dripper from the cup',
+             'Remove the coffee grounds and paper filter from the dripper',
+             'Discard the coffee grounds and paper filter',
+             'finished']
+
+#class_str = ['background', 'step1', 'step2', 'step3', 'step4']
```
Comment (Purg) [inline angel_system/activity_hmm/simulate_data.py:55]: dead code / config.

File: angel_system/activity_hmm/simulate_data.py [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+    import matplotlib.pyplot as plt
+    from matplotlib.ticker import MultipleLocator, FormatStrFormatter, \
+        AutoMinorLocator
+    HAS_MATLOTLIB = True
+except ModuleNotFoundError:
+    HAS_MATLOTLIB = False
+
+
+
+class_str = ['background',
```
Comment (Purg) [inline angel_system/activity_hmm/simulate_data.py:26]: Would it be good to read from the config file now?

File: angel_system/activity_hmm/simulate_data.py [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+num_trials = 1000
+tp_hmm = []
+fp_hmm = []
+for i in range(num_trials):
+    print('%i/%i' % (i + 1, num_trials))
+
+    # Simulate trial
+    X0, Z0, X0_, Z0_ = model.sample(1000)[1:]
+
+    if False:
```
Comment (Purg) [inline angel_system/activity_hmm/simulate_data.py:359]: Dead code-path.

File: angel_system/activity_hmm/simulate_data.py [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+
+    X = X0[inds]
+    Z = Z0[inds]
+
+    s = get_skip_score(model, model_skip, X)
+
+    tp_hmm.append(s)
+
+
+if False:
```
Comment (Purg) [inline angel_system/activity_hmm/simulate_data.py:395]: Dead code-path.

**Review by Purg — COMMENTED** [review]
Just some minor comments, and one or two improvement suggestions

File: angel_system/activity_hmm/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+from .core import *
```
[the remark this answers is not in the document]
Comment (mattbrown11) [reply angel_system/activity_hmm/__init__.py]: done

File: angel_system/activity_hmm/core.py [hunk 6]
```diff
@@ -0,0 +1,773 @@
+import numpy as np
+from hmmlearn.hmm import GaussianHMM
```
[the remark this answers is not in the document]
Comment (mattbrown11) [reply angel_system/activity_hmm/core.py:2]: yeah, that would probably be best for now (i.e., me not doing it).

File: angel_system/activity_hmm/simulate_data.py (hunk 1 above)
Comment (mattbrown11) [reply angel_system/activity_hmm/simulate_data.py]: moved to a sandboxy scripts subdir

File: angel_system/activity_hmm/simulate_data.py (hunk 2 above)
Comment (mattbrown11) [reply angel_system/activity_hmm/simulate_data.py]: moved to a sandboxy scripts subdir

File: angel_system/activity_hmm/simulate_data.py (hunk 3 above)
Comment (mattbrown11) [reply angel_system/activity_hmm/simulate_data.py]: moved to a sandboxy scripts subdir

File: angel_system/activity_hmm/simulate_data.py (hunk 4 above)
Comment (mattbrown11) [reply angel_system/activity_hmm/simulate_data.py]: moved to a sandboxy scripts subdir

File: angel_system/activity_hmm/core.py [hunk 7]
```diff
@@ -0,0 +1,773 @@
...
+            Path to the configuration yaml.
+
+        """
+        with open(config_fname, 'r') as stream:
+            config = yaml.safe_load(stream)
+
+        default_mean_conf = config['hmm']['class_mean_conf']
+        default_std_conf = config['hmm']['class_std_conf']
+
+        self.dt = 0.5
```
[the remark this answers is not in the document]
Comment (mattbrown11) [reply angel_system/activity_hmm/core.py]: Done

File: angel_system/activity_hmm/core.py [hunk 8]
```diff
@@ -0,0 +1,773 @@
...
+            class_str = []
+            med_class_duration = []
+            class_mean_conf = []
+            class_std_conf = []
+        else:
+            # Start with a background class.
+            class_str = ['background']
+            med_class_duration = [1]
+            class_mean_conf = [default_mean_conf]
+            class_std_conf = [default_std_conf]
```
[the remark this answers is not in the document]
Comment (mattbrown11) [reply angel_system/activity_hmm/core.py]: Done

File: angel_system/activity_hmm/core.py [hunk 9]
```diff
@@ -0,0 +1,773 @@
...
+                                '%i but it has \'id\' %i' %
+                                (i, config_fname, i, ii))
+
+            class_str.append(config['steps'][i]['description'])
+            med_class_duration.append(config['steps'][i]['median_duration_seconds'])
+
+            try:
+                class_mean_conf.append(config['steps'][i]['class_mean_conf'])
+            except KeyError:
+                class_mean_conf.append(default_mean_conf)
```
[the remark this answers is not in the document]
Comment (mattbrown11) [reply angel_system/activity_hmm/core.py:62]: Added check

File: angel_system/activity_hmm/core.py [hunk 10]
```diff
@@ -0,0 +1,773 @@
...
+                continue
+
+            model.startprob_[i] = 1
+
+        model.startprob_ /= np.sum(model.startprob_)
+
+        # Indices into N_-length arrays (e.g., class_str_) that correspond to the
+        # original N-length arrays (e.g., class_str).
+        not_bckg_mask_ = ~bckg_mask.copy()
+        #not_bckg_mask_[0] = True
```
[the remark this answers is not in the document]
Comment (mattbrown11) [reply angel_system/activity_hmm/core.py]: fixed

File: angel_system/activity_hmm/core.py [hunk 11]
```diff
@@ -0,0 +1,773 @@
...
+
```
[the remark this answers is not in the document]
Comment (mattbrown11) [reply angel_system/activity_hmm/core.py:279]: better?

**Review by Purg — APPROVED** [review]

**Pull request merged**